templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -25,7 +25,8 @@ contents:
             logger -s "NM resolv-prepender: Prepending 'nameserver $NAMESERVER_IP' to /etc/resolv.conf (other nameservers from /var/run/NetworkManager/resolv.conf)"
             sed -e "/^search/d" \
                 -e "/Generated by/c# Generated by KNI resolv prepender NM dispatcher script\nsearch $DOMAIN\nnameserver $NAMESERVER_IP" \
-                /var/run/NetworkManager/resolv.conf > /etc/resolv.conf
+                /var/run/NetworkManager/resolv.conf > /etc/resolv.tmp
+            mv -f /etc/resolv.tmp /etc/resolv.conf
         else
             logger -s "NM resolv-prepender: Couldn't find a Virtual IP, just updating resolv.conf"
             cp /var/run/NetworkManager/resolv.conf /etc/resolv.conf
```

Comment:
`cp` is also not atomic - this is a really big trap that a whole lot of people have fallen into.  You could probably unify these by having the second path do
`cp /var/run/NetworkManager/resolv.conf /etc/resolv.tmp`
and then running
`mv -f /etc/resolv.tmp /etc/resolv.conf` after the `if`.

Reply:
Good idea. Will do.